src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs

```diff
@@ -639,6 +636,16 @@ private static FormulaValue SortOptionSet(List<(DValue<RecordValue> row, Formula
             });
 
             return new InMemoryTableValue(irContext, pairs.Select(pair => pair.row));
+        }        
+
+        private static async ValueTask<FormulaValue> Refresh(EvalVisitor runner, EvalVisitorContext context, IRContext irContext, FormulaValue[] args)
+        {
+            if (args[0] is IRefreshable r)
+            {
+                r.Refresh();
+            }
+
+            return FormulaValue.NewBlank();

```

Comment:
>return FormulaValue.NewBlank(); [](http://example.com/codeflow?start=12&length=31)

To consider: should this be an error?

Ideally, we'd get compile time errors when passing a non-refreshable item, but I don't think we have the infrastructure for that.

Reply:
We should have a runtime error - that would match the Power Apps behavior. We don't always get compile-time errors when we have a non-refreshable item, like in the example below:

    Set(cache, FirstNSharePointList, 10));
    Refresh(If(1<0, SharePointList, cache))

Reply:
The documentation doesn't meantion any possible error.
[https://learn.microsoft.com/en-us/power-platform/power-fx/reference/function-refresh](https://learn.microsoft.com/en-us/power-platform/power-fx/reference/function-refresh)

PA Client returns "only managed connections can be refreshed".

We can do the same to be consistent.